### ISSUE TITLE
Parquet PyArrow tests failing

### ISSUE THREAD

**Issue by mrocklin** [dask/dask#3011]
https://travis-ci.org/dask/dask/jobs/318270245

There are several failing tests.  Here is the output from one:

```
____________________________ test_parquet[pyarrow] _____________________________

[gw1] linux2 -- Python 2.7.14 /home/travis/miniconda/envs/test-environment/bin/python


s3 = <s3fs.core.S3FileSystem object at 0x7f6353d3dd10>, engine = 'pyarrow'


    @pytest.mark.parametrize("engine", ['pyarrow', 'fastparquet'])

    def test_parquet(s3, engine):

        dd = pytest.importorskip('dask.dataframe')

        pytest.importorskip(engine)

        import pandas as pd

        import numpy as np

    

        url = 's3://%s/test.parquet' % test_bucket_name

    

        data = pd.DataFrame({'i32': np.arange(1000, dtype=np.int32),

                             'i64': np.arange(1000, dtype=np.int64),

                             'f': np.arange(1000, dtype=np.float64),

                             'bhello': np.random.choice(

                                 [u'hello', u'you', u'people'],

                                 size=1000).astype("O")},

                            index=pd.Index(np.arange(1000), name='foo'))

        df = dd.from_pandas(data, chunksize=500)

        df.to_parquet(url, engine=engine)

    

        files = [f.split('/')[-1] for f in s3.ls(url)]

        assert '_common_metadata' in files

        assert 'part.0.parquet' in files

    

>       df2 = dd.read_parquet(url, index='foo', engine=engine)


dask/bytes/tests/test_s3.py:309: 

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

dask/dataframe/io/parquet.py:558: in read_parquet

    categories=categories, index=index)

dask/dataframe/io/parquet.py:320: in _read_pyarrow

    meta = _meta_from_dtypes(all_columns, schema.names, dtypes, index_cols)

dask/dataframe/io/parquet.py:26: in _meta_from_dtypes

    df = meta[list(to_read_columns)]

../../../miniconda/envs/test-environment/lib/python2.7/site-packages/pandas/core/frame.py:1958: in __getitem__

    return self._getitem_array(key)

../../../miniconda/envs/test-environment/lib/python2.7/site-packages/pandas/core/frame.py:2002: in _getitem_array

    indexer = self.loc._convert_to_indexer(key, axis=1)

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 


self = <pandas.core.indexing._LocIndexer object at 0x7f6353e6aa50>

obj = ['bhello', 'f', 'i32', 'i64', '__index_level_0__', 'foo'], axis = 1

is_setter = False


    def _convert_to_indexer(self, obj, axis=0, is_setter=False):

        """

            Convert indexing key into something we can use to do actual fancy

            indexing on an ndarray

    

            Examples

            ix[:5] -> slice(0, 5)

            ix[[1,2,3]] -> [1,2,3]

            ix[['foo', 'bar', 'baz']] -> [i, j, k] (indices of foo, bar, baz)

    

            Going by Zen of Python?

            'In the face of ambiguity, refuse the temptation to guess.'

            raise AmbiguousIndexError with integer labels?

            - No, prefer label-based indexing

            """

        labels = self.obj._get_axis(axis)

    

        if isinstance(obj, slice):

            return self._convert_slice_indexer(obj, axis)

    

        # try to find out correct indexer, if not type correct raise

        try:

            obj = self._convert_scalar_indexer(obj, axis)

        except TypeError:

    

            # but we will allow setting

            if is_setter:

                pass

    

        # see if we are positional in nature

        is_int_index = labels.is_integer()

        is_int_positional = is_integer(obj) and not is_int_index

    

        # if we are a label return me

        try:

            return labels.get_loc(obj)

        except LookupError:

            if isinstance(obj, tuple) and isinstance(labels, MultiIndex):

                if is_setter and len(obj) == labels.nlevels:

                    return {'key': obj}

                raise

        except TypeError:

            pass

        except (ValueError):

            if not is_int_positional:

                raise

    

        # a positional

        if is_int_positional:

    

            # if we are setting and its not a valid location

            # its an insert which fails by definition

            if is_setter:

    

                # always valid

                if self.name == 'loc':

                    return {'key': obj}

    

                # a positional

                if (obj >= self.obj.shape[axis] and

                        not isinstance(labels, MultiIndex)):

                    raise ValueError("cannot set by positional indexing with "

                                     "enlargement")

    

            return obj

    

        if is_nested_tuple(obj, labels):

            return labels.get_locs(obj)

    

        elif is_list_like_indexer(obj):

    

            if is_bool_indexer(obj):

                obj = check_bool_indexer(labels, obj)

                inds, = obj.nonzero()

                return inds

            else:

    

                # Have the index compute an indexer or return None

                # if it cannot handle

                indexer, objarr = labels._convert_listlike_indexer(

                    obj, kind=self.name)

                if indexer is not None:

                    return indexer

    

                # unique index

                if labels.is_unique:

                    indexer = check = labels.get_indexer(objarr)

    

                # non-unique (dups)

                else:

                    (indexer,

                     missing) = labels.get_indexer_non_unique(objarr)

                    # 'indexer' has dupes, create 'check' using 'missing'

                    check = np.zeros_like(objarr)

                    check[missing] = -1

    

                mask = check == -1

                if mask.any():

>                   raise KeyError('%s not in index' % objarr[mask])

E                   KeyError: "['foo'] not in index"


../../../miniconda/envs/test-environment/lib/python2.7/site-packages/pandas/core/indexing.py:1231: KeyError

------------------------------ Captured log call -------------------------------

core.py                   1242 DEBUG    Fetch: test/test.parquet/part.0.parquet, 0-5255312

core.py                   1242 DEBUG    Fetch: test/test.parquet/part.0.parquet, 0-5255312

core.py                   1242 DEBUG    Fetch: test/test.parquet/part.1.parquet, 0-5255374

```

I apologize for the poor copy-paste formatting.  I recommend going to the travis-ci link. 

cc @wesm @xhochy

**Comment by jcrist** [comment]
This is due the recent release of pyarrow, and #2901. #2973 is intended to fix this (cc @TomAugspurger).

**Comment by mrocklin** [comment]
Ah, thanks for pointing me to the right issue

**Comment by jakirkham** [comment]
> I apologize for the poor copy-paste formatting. I recommend going to the travis-ci link.

Yeah have generally found copy-paste from Travis CI logs to not be aesthetically pleasing and have had to manually fix it via `ipython` (e.g. `%paste s` and then `print(s[::2])`). Finally got around to letting them know ( https://github.com/travis-ci/travis-ci/issues/8942 ). 😄

**Comment by jcrist** [comment]
Fixed by #2973. Closing.
